agent/structs/connect_proxy_config.go

```diff
@@ -303,7 +312,11 @@ type Upstream struct {
 
 	// LocalBindPort is the ip address a side-car proxy should listen on for traffic
 	// destined for this upstream service. Required.
-	LocalBindPort int `alias:"local_bind_port"`
+	LocalBindPort int `json:",omitempty" alias:"local_bind_port"`
+
+	// These are exclusive with LocalBindAddress/LocalBindPort
+	LocalBindSocketPath string `json:",omitempty" alias:"local_bind_socket_path"`
+	LocalBindSocketMode uint32 `json:",omitempty" alias:"local_bind_socket_mode"`
```

Comment:
Due to the number of round trips involved in threading this needle, I suspect we might want to keep it in its string form all the way until use-time in `agent/xds` (but validate it up front). That way any API queries that may dump these various places it shows up still show the mode as the user thought they configured it, not the numeric base-10 variation.

Reply:
Followed your suggestion here, and converted to string,